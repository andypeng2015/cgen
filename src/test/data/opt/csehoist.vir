module foo

export function w $foo(w %x, w %y) {
@start:
  %c = eq.w %x, 0_w
  br %c, @zero, @nonzero
@zero:
  %z = add.w %y, 1_w
  call $do_something(%z)
  jmp @done
@nonzero:
  jmp @done
@done:
  %w = add.w %y, 1_w
  ret %w
}
