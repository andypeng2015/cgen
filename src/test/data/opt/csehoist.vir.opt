module foo

export function w $foo(w %x, w %y) {
@2:
  %2 = add.w %y, 0x1_w ; @12
  %3 = eq.w %x, 0x0_w ; @13
  br %3, @3, @6
@3:
  call $do_something(%2) ; @8
  jmp @6
@6:
  ret %2
}
