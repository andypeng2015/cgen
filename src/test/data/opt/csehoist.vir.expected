module foo

export function w $foo(w %x, w %y) {
@0:
  %1 = eq.w %x, 0x0_w ; @9
  %0 = add.w %y, 0x1_w ; @8
  br %1, @1, @4
@1:
  call $do_something(%0) ; @6
  jmp @4
@4:
  ret %0
}
